api/src/Repository/RepositoryInterface.php

```diff
@@ -0,0 +1,28 @@
+<?php declare(strict_types=1);
+/*.
+    require_module 'standard';
+.*/
+
+namespace App\Repository;
+
+interface RepositoryInterface
+{
+
+
+    public function insert(/*.mixed.*/$data): bool;
+
+
+    public function selectAll(): array;
+
+
+    public function selectById(/*.mixed.*/$id): array;
```

Comment:
Thinking about this a little bit more now as well.

Would it make sense to maybe have `selectById` where it's one ID, and `selectByIds` where it could be multiple IDs? In the implementing class we could extract logic currently in `selectById` to a private function that is called by each of those instead.

It would allow us to be a bit more explicit in the code as well, rather than having to keep track of whether we're dealing with one ID or multiple when calling into the repository layer. What are your thoughts on this?

Reply:
I can go either way. I am not passionate about one or the other.